### PR TITLE
getSpaceServerSide moved to frontend

### PR DESCRIPTION
**Pull request by MSamiTariq**
# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [ ] I have carefully reviewed my code and believe it is final from my side.
- [ ] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [ ] I have tested the new feature or updates in both light and dark themes.
- [ ] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [ ] I have tested my changes on mobile screen sizes.
- [ ] I have tested my changes on tablet screen sizes.
- [ ] I have tested my changes on laptop screen sizes.

### Functional Testing
- [ ] I have thoroughly tested the functionality of my changes.
- [ ] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [ ] Explain the task/bug you are solving
- [ ] Show the changes in the UI
- [ ] Show the changes in the codebase
- [ ] Show the changes you tested to make sure it works
- [ ] Show the changes in various screen sizes

### CODE REVIEW

File: academy-ui/prisma/migrations/20240702162613_initial_migration/migration.sql [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+-- CreateTable
+CREATE TABLE "crypto_login_nonce" (
+    "user_id" TEXT NOT NULL,
+    "nonce" TEXT NOT NULL,
+    "expires" TIMESTAMP(3) NOT NULL
+);
+
+-- CreateTable
+CREATE TABLE "spaces" (
```
Comment (RobinNagpal) [inline academy-ui/prisma/migrations/20240702162613_initial_migration/migration.sql:60]: We can limit this to just space entity

**Comment by RobinNagpal** [comment]
```
/api/spaces 
    route.ts
-> Check if domain is passed as search parameter, then find and return spaces by domain
        -> Else return 400

/api/spaces/:spaceId
    route.ts
        -> Find and return space by spaceId
        -> Else return 400


/api/spaces/:spaceId/bytes
    route.ts

/api/spaces/:spaceId/bytes/:byteId
    route.ts


/api/spaces/:spaceId/bytes/:byteId/ratings
    route.ts

/api/spaces/:spaceId/bytes/:byteId/ratings/:ratingId
    route.ts

/api/spaces/:spaceId/guides
    route.ts

/api/spaces/:spaceId/guides/:guideId
    route.ts
```

File: academy-ui/src/utils/space/getSpaceServerSide.tsx [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
 
-  const response = await axios.get(process.env.V2_API_SERVER_URL?.replace('/graphql', '') + '/extended-space', {
+  const response = await axios.get(process.env.ACADEMY_UI_SERVER_URL + '/api/spaces', {
     params: {
       domain: host!,
     },
   });
 
-  const space = response?.data as SpaceWithIntegrationsFragment;
+  const space = response?.data.body[0] as SpaceWithIntegrationsFragment;
```
Comment (MSamiTariq) [inline academy-ui/src/utils/space/getSpaceServerSide.tsx:14]: Since the current route `/api/spaces` returns a list of spaces, that's why I'm referencing the first element of the array here. Should I create a separate route like `/api/domain/:domainName`, or is it fine?

File: academy-ui/src/app/api/helpers/space/upsertSpaceIntegrations.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export async function upsertSpaceIntegrations(spaceInput: UpsertSpaceInput, user: DoDaoJwtTokenPayload) {
+  await prisma.spaceIntegration.upsert({
+    create: {
+      id: v4(),
+      spaceId: spaceInput.id,
+      academyRepository: spaceInput.spaceIntegrations.academyRepository,
+      discordGuildId: spaceInput.spaceIntegrations.discordGuildId,
+      gitGuideRepositories: spaceInput.spaceIntegrations.gitGuideRepositories,
+      gnosisSafeWallets: JSON.stringify(spaceInput.spaceIntegrations.gnosisSafeWallets),
```
Comment (MSamiTariq) [inline academy-ui/src/app/api/helpers/space/upsertSpaceIntegrations.ts:14]: This doesn't work without json.stringify. Need to verify if this works correctly
